apps/qa/src/pages/ingest/ingest.py

```diff
@@ -1,21 +1,206 @@
 def ingest():
     import streamlit as st
-    from .helpers import archive_raw_data
+    from .helpers import archive_raw_data, dummy_library_call, dummy_archive_raw_data
     from pathlib import Path
+    import time
 
-    st.title("Ingest Dataset")
-
-    dataset_name = st.text_input("Dataset Name")
-    version = st.text_input("Version")
-    uploaded_file = st.file_uploader("Choose a file")
-    s3_path = Path("inbox") / dataset_name / version
+    if "ingest" not in st.session_state:
+        st.session_state["ingest"] = {
+            "running": False,
+            "upload_status": None,
+            "library_status": None,
+            "error_message": None,
+            "retry": False,
+            "s3_path": None,
+        }
 
-    ingest_button_pressed = st.button("Ingest")
-    if ingest_button_pressed == True:
+    def lock_for_ingest():
         if dataset_name and version and uploaded_file:
```

Comment:
not sure how mypy isn't erroring here - these aren't defined at this point, so this is relying on these variables happening to exist. They should be provided as arguments to this function (and similar functions below)

Reply:
Can you add type signatures for the functions as well? And return as well (even though it's just `-> None`)

Reply:
Ah that's why mypy wasn't catching this - mypy doesn't lint untyped functions

Reply:
In that case, add a type signature for the top-level function as well - this is a pattern we should probably enforce for other pages, but that's obviously out of scope here